### PR TITLE
Flash simulator stub

### PR DESCRIPTION
**Pull request by de-nordic**
The PR contains two commits:
 - implementing the flash simulator stub, a device that always read erase_value and does nothing on write/erase; the device still respects offset, size and other parameters of flash simulator as set in a DTS;
 - tests

### CODE REVIEW

File: drivers/flash/Kconfig.simulator [hunk 1]
```diff
@@ -26,6 +26,13 @@ config FLASH_SIMULATOR_DOUBLE_WRITES
 	 If selected, writing to a non-erased program unit will succeed, otherwise, it will return an error.
 	 Keep in mind that write operations can only pull bits to zero, regardless.
 
+config FLASH_SIMULATOR_STUB
```
Comment (MarekPieta) [inline drivers/flash/Kconfig.simulator:29]: Maybe we could select variant of flash simulator using a choice?
That could make it easier to understand. We could also select proper source file relying on selected choice option

Comment (MarekPieta) [reply drivers/flash/Kconfig.simulator:29]: The choice symbol could also be used for Kconfig dependencies below

File: tests/drivers/flash_simulator_stub/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA.
```
Comment (MarekPieta) [inline tests/drivers/flash_simulator_stub/CMakeLists.txt:2]: 2021

File: tests/drivers/flash_simulator_stub/boards/qemu_x86_ev_0x00.overlay [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&sim_flash {
+	erase-value = < 0x00 >;
```
Comment (MarekPieta) [inline tests/drivers/flash_simulator_stub/boards/qemu_x86_ev_0x00.overlay:8]: why 0x00 ?

Comment (de-nordic) [reply tests/drivers/flash_simulator_stub/boards/qemu_x86_ev_0x00.overlay:8]: Because this test variant is for 0x00 erased flash.

**Comment by Laczen** [comment]
I don't understand the usecase you mentioned in the previous PR.

Since a configuration change is needed anyhow you could just disable the flash-sim completely to reclaim the memory.

But then again, maybe there is something I am missing.

**Comment by de-nordic** [comment]
Implemented as other device driver.

**Pull request closed without merging**